tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,523 @@
+import dataclasses
+import datetime
+import os
+import sys
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+import click
+
+import tmt
+import tmt.options
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt.utils import ProvisionError, updatable_message
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+import asyncio
+from functools import wraps
+
+mrack = Any
+providers = Any
+NotAuthenticatedError = Any
+BEAKER = Any
+BeakerProvider = Any
+BeakerTransformer = Any
+TmtBeakerTransformer = Any
+
+
+def import_and_load_mrack_deps() -> None:
+    """
+    Import mrack module only when needed
+
+    Until we have a separate package for each plugin.
+    """
+    global mrack
+    global providers
+    global NotAuthenticatedError
+    global BEAKER
+    global BeakerProvider
+    global BeakerTransformer
+    global TmtBeakerTransformer
+
+    try:
+        import mrack
+        from mrack.errors import NotAuthenticatedError
+        from mrack.providers import providers
+        from mrack.providers.beaker import PROVISIONER_KEY as BEAKER
+        from mrack.providers.beaker import BeakerProvider
+        from mrack.transformers.beaker import BeakerTransformer
+
+        providers.register(BEAKER, BeakerProvider)
+
+    except ImportError:
+        raise ProvisionError(
+            "Install 'mrack' to provision using this method.")
+
+    class TmtBeakerTransformer(BeakerTransformer):  # type: ignore[misc]
+        def _parse_amount(self, in_string: str) -> Tuple[str, int]:
+            chunks = in_string.split(" ")
+
+            amount: int = 0
+            operator: str = "="
+            for chunk in chunks:
+                try:
+                    amount = int(chunk)
+                except ValueError:
+                    if chunk.isupper():
+                        amount = int(amount * size_translation[chunk])
+                    else:
+                        operator = chunk
+
+            return operator, amount
+
+        def _translate_tmt_hw(self, hw: Dict[str, Any]) -> Dict[str, Any]:
+            key = "_key"
+            value = "_value"
+            op = "_op"
+
+            system = {}
+            disks = []
+            cpu = {}
+
+            for key, val in hw.get("constraints", {}).items():
```

Comment:
There will be no `constraints`, that's Artemis API relict. In tmt nodes, this is what you'd get:

```yaml
provision:
  how: ...
  hardware:
    cpu:
      processors: >= 2
    memory: >= 8 GiB
```

Therefore no extra layer under `hw`.